front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -177,9 +177,24 @@ export const usePathfinding = (
 
   const { updatePathSteps } = useOsrdConfActions();
 
+  const [invalidItems, setInvalidItems] = useState<string[]>([]);
+
   const generatePathfindingParams = (): PostInfraByInfraIdPathfindingBlocksApiArg | null => {
     setPathProperties?.(undefined);
-    return getPathfindingQuery({ infraId, rollingStock, origin, destination, pathSteps });
+
+    const filteredPathSteps = pathSteps.filter(
+      (step) =>
+        step !== null &&
+        step.coordinates !== null &&
+        !('trigram' in step && invalidItems.includes(step.trigram))
```

Comment:
I think this is enough :)
```suggestion
        pathStepIsInvalid(step, invalidItems)
```